src/ChatApp.WebApi/Agents/CreativeWriterSession.cs

```diff
@@ -2,48 +2,31 @@
 // Licensed under the MIT License.
 
 using ChatApp.ServiceDefaults.Contracts;
-using ChatApp.WebApi.Model;
 using Microsoft.SemanticKernel;
 using Microsoft.SemanticKernel.Agents;
 using Microsoft.SemanticKernel.Agents.Chat;
-using Microsoft.SemanticKernel.Connectors.AzureOpenAI;
+using Microsoft.SemanticKernel.Agents.AzureAI;
 using System.Text;
 
 namespace ChatApp.WebApi.Agents;
 
-public class CreativeWriterSession(Kernel kernel, Kernel bingKernel, Kernel vectorSearchKernel)
+public class CreativeWriterSession(Kernel kernel, Azure.AI.Projects.AgentsClient agentsClient, AzureAIAgent researcherAgent, ChatCompletionAgent marketingAgent, ChatCompletionAgent writerAgent, ChatCompletionAgent editorAgent)
 {
-    private const string ResearcherName = "Researcher";
-    private const string MarketingName = "Marketing";
-    private const string WriterName = "Writer";
-    private const string EditorName = "Editor";
 
     internal async IAsyncEnumerable<AIChatCompletionDelta> ProcessStreamingRequest(CreateWriterRequest createWriterRequest)
     {
-        ChatCompletionAgent researcherAgent = new(ReadFileForPromptTemplateConfig("./Agents/Prompts/researcher.yaml"))
-        {
-            Name = ResearcherName,
-            Kernel = bingKernel,
-            Arguments = CreateFunctionChoiceAutoBehavior(),
-            LoggerFactory = bingKernel.LoggerFactory
-        };
-
-        ChatCompletionAgent marketingAgent = new(ReadFileForPromptTemplateConfig("./Agents/Prompts/marketing.yaml"))
-        {
-            Name = MarketingName,
-            Kernel = vectorSearchKernel,
-            Arguments = CreateFunctionChoiceAutoBehavior(),
-            LoggerFactory = vectorSearchKernel.LoggerFactory
-        };
+        // create an conversation Thread with the Researcher agent
+        Azure.Response<Azure.AI.Projects.AgentThread> threadResponse = await agentsClient.CreateThreadAsync();
```

Comment:
Creating a new conversation thread on purpose for every interaction, to forget the conversation history as this is not intended by this example.